src/tainting/Dataflow_tainting.ml

```diff
@@ -772,7 +796,11 @@ and check_tainted_lval_aux env (lval : IL.lval) :
         | (`None | `Tainted _) as st -> (
             match Lval_env.dumb_find lval_env lval with
             | (`Clean | `Tainted _) as st' -> st'
-            | `None -> st)
+            | `None ->
+                (* HACK(field-sensitivity): Java: If we encounter `obj.getX` and `obj` has
+                   * polymorphic taint,  and we know nothing specific about `obj.getX`, then
+                   * we add the same offset `.getX` to the polymorphic taint coming from `obj`. *)
+                add_offset_to_poly_taint_in_st env lval st)
```

Comment:
I think I'm missing something. Where do we resolve `setX` vs `setY` to know that `setX` only applies to `x`?

Reply:
That was done as part of a previous PR, you reviewed it too I think. There we intercept `obj.setX(foo)` and we propagate the taint of `foo` into `obj.getX`. I made this more clear here I hope:
```ocaml
        let lval_env =
          (* HACK: Java: If we encounter `obj.setX(arg)` we interpret this as `obj.getX = arg`. *)
          propagate_taint_via_java_setter {env with lval_env} e args all_args_taints
        in
```